gpytorch/priors/lkj_prior.py

```diff
@@ -61,6 +61,58 @@ def log_prob(self, X):
         log_diag_sum = psd_safe_cholesky(X, upper=True).diagonal(dim1=-2, dim2=-1).log().sum(-1)
         return self.C + (self.eta - 1) * 2 * log_diag_sum
 
+    def _rsample(self, sample_shape=None, return_covariance=False, return_cholesky=False):
+        # this sampling method comes from
+        # https://github.com/rmcelreath/rethinking/blob/2acf2fd7b01718cf66a8352c52d001886c7d3c4c/R/distributions.r#L214
+        if sample_shape is None:
+            sample_shape = torch.Size()
+
+        N = self.n.item()
+
+        alpha = self.eta + (N - 1) / 2
+        r12 = 2.0 * Beta(alpha, alpha).rsample(sample_shape) - 1
+        R = torch.zeros(*sample_shape, N, N, device=self.eta.device, dtype=self.eta.dtype)
+
+        R[..., 0, 0] = 1.0
+        R[..., 0, 1] = r12
+        R[..., 1, 1] = (1 - r12.pow(2)).pow(0.5)
+
+        for m in range(1, N):
+            alpha = alpha - 0.5
+            y = Beta(m / 2, alpha).rsample(sample_shape)
+
+            # uniform on a hypersphere
+            z = torch.randn(*sample_shape, m, device=self.eta.device, dtype=self.eta.dtype)
+            z = z / z.pow(2.0).sum(-1, keepdim=True).pow(0.5)
+
+            R[..., :m, m] = y.pow(0.5).unsqueeze(-1) * z
+            R[..., m, m] = (1.0 - y).pow(0.5)
+
+        # the sample R is drawn from as the cholesky of a covariance matrix, not a correlation
+        # matrix, so we may need to re-compute a cholesky decomposition
+        if return_covariance:
+            if return_cholesky:
+                return R
+            else:
+                return R.matmul(R.transpose(-1, -2))
+        else:
+            sample_as_covariance = R.matmul(R.transpose(-1, -2))
+
+            # we need to back-convert the resulting covariance matrix into a correlation matrix
+            inverse_sqrt_diagonal = torch.diagonal(sample_as_covariance, dim1=-2, dim2=-1).clamp(min=1e-6).pow(-0.5)
+            correlation_mat = (
+                inverse_sqrt_diagonal.unsqueeze(-1) * sample_as_covariance * inverse_sqrt_diagonal.unsqueeze(-2)
+            )
+
+            # and then return the cholesky of the correlation matrix
+            if return_cholesky:
+                return correlation_mat.cholesky()
+            else:
+                return correlation_mat
+
+    def rsample(self, sample_shape=torch.Size([])):
```

Comment:
no need to instantiate empty lists for this
```suggestion
    def rsample(self, sample_shape=torch.Size()):
```